gems/sorbet-runtime/test/types/props/constructor.rb

```diff
@@ -145,4 +145,76 @@ def initialize(hash={})
     end
     assert_equal("Missing required prop `name` for class `Opus::Types::Test::Props::ConstructorTest::CustomInitializeStruct`", err.message)
   end
+
+  class ComplexStruct < T::Struct
+    prop :primitive, Integer
+    prop :nilable, T.nilable(Integer)
```

Comment:
This is implicitly relying on other tests to verify that initialization is handled correctly, is that correct?

I think there needs to be a `T.deprecated_enum` type in here somewhere.